cognite/client/data_classes/data_modeling/views.py

```diff
@@ -863,3 +866,83 @@ def dump(self, camel_case: bool = True) -> dict[str, Any]:
             output["connection_type"] = "multi_reverse_direct_relation"
 
         return output
+
+
+@dataclass
+class PropertyUnit(CogniteObject):
+    property: str
+    unit: UnitReference | UnitSystemReference
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {"property": self.property, "unit": self.unit.dump(camel_case)}
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> PropertyUnit:
+        return cls(
+            property=resource["property"],
+            unit=UnitReference.load(resource["unit"])
+            if "externalId" in resource["unit"]
+            else UnitSystemReference.load(resource["unit"]),
+        )
+
+
+class SourceCore(CogniteObject):
```

Comment:
I don't think this serves much purpose as is. Why not make this the superclass of SourceSelector?